app/controllers/palettes_controller.rb

```diff
@@ -0,0 +1,49 @@
+class PalettesController < ApplicationController
+
+  def index
+    @palettes = Palette.all
+      render json: Palette.all
+    end
+  end
+
+  def show
+    @palette = get_palette
+      render :get_palette
+    end
+
+  def create
+    @palette = Palette.new(palette_params)
+      if @palette.save
+        render status: :created 
+      else
+        render status: :unprocessable_entity
+      end
+    end
+
+  def update
+    @palette = get_palette
+      if @palette.update(palette_params)
+        head :no_content
```

Comment:
you don't want to use head :no_content here, because you want it to return something if it is updated. so you will still need to say format.json { render json: @palette }